### ISSUE TITLE
As an Open data Hub manager I want a better and simpler navigation

### ISSUE THREAD

**Issue by gappc** [noi-techpark/it.bz.opendatahub.databrowser#204]
The current Data Browser menu entries contain all endpoints of the ODH tourism domain from ~ July 2022.
Some entries are not of interest for the normal user (e.g. ODHTags).
We need to define which entries the menu should contain.

The next tasks are:
- [x] @pkritzinger analyzes the actual navigation
- [x] @pkritzinger collects feedback from the first beta users
- [x] @pkritzinger reviews the menu entry and makes a new proposal
- [x] the gorup reviews and agrees the final proposal 
- [x] @gappc implements the new menu

**Comment by d3liaa** [comment]
I wanted also to add.
The menu contains entries such as Activities, Activities and Points of Interest and  Points of Interest. 
From these only "Activities and Points of Interest" should exist as is the only one with a respective endpoint the others do not exist or are deprecated.

**Comment by sseppi** [comment]
I added this issue the the Mileston 2.2 since it is strongly connected to the issue #278 which aims to improve the navigation of the entire data browser (e.g. header, footer, connection with other Open Data Hub tools, metadata API, etc.)

**Comment by d3liaa** [comment]
I reviewed the menu and created a new proposal. Maybe if @pkritzinger has time he can review it and decide if it needs further modifications.

New menu proposal:

 Locations
	-   [District](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/District)
	-   [Experience Area](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/ExperienceArea)
	-   [MetaRegion](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/MetaRegion)
	-   [Municipality](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Municipality)
	-   [Region](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Region)
	-   [Ski Area](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/SkiArea)
	-   [Ski Region](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/SkiRegion)
	-   [Tourism Association List](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/TourismAssociation)
	-   [Venue](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Venue)
	-   [Webcams](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/WebcamInfo)

Events
	-   [NOI Events](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/EventShort)
	-   [Events](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Event)

Weather
	-   [Measuring Point](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Weather/Measuringpoint)
	-   [Snow Report](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Weather/SnowReport)
	-  [Weather](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Weather)
	-   [Weather District](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Weather/District)
	-   [Weather in Realtime](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Weather/Realtime)
	-   [WeatherInfo](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/WeatherHistory)

Accommodation
	-   [Accommodation](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Accommodation)

Activities and Points of Interest
	-   [Activities and Points of Interest (POIs)](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/ODHActivityPoi)

Articles 
	-   [Articles](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/Article)

WineAward
	-   [WineAward](https://databrowser.opendatahub.testingmachine.eu/dataset/table/tourism/v1/WineAward)

**Comment by sseppi** [comment]
Hi Delia,
thank you for your proposal which is inline with my thoughts.
In my opinion we should try to auto-generate also the part of the menu related to the access of the datasets.
I would suggest to have a main menu point for each "Endpoint" and a sub menu point for each "Dataset" related to an Endpoint.
Using this logic should be possible to autogenerate the menu out of the Metadata API?

**Comment by gappc** [comment]
@sseppi that seems reasonable.

@RudiThoeni what do you say? Which fields could be used to implement such a logic, e.g. `ApiIdentifier` as main menu point and then `Shortname` as sub menu point?

**Comment by RudiThoeni** [comment]
@gappc 
I can create custom fields if you need, no problem.
ApiIdentifier has to be exactly the Endpoint Name on the api, Shortname can be set whatever we want. If this is ok we can use this fields....... otherwise no problem if i add a field "Parentmenu" or something, just let me know and i will add it

**Comment by sseppi** [comment]
> @gappc I can create custom fields if you need, no problem. ApiIdentifier has to be exactly the Endpoint Name on the api, Shortname can be set whatever we want. If this is ok we can use this fields....... otherwise no problem if i add a field "Parentmenu" or something, just let me know and i will add it

@gappc @RudiThoeni In my point of view is fine to use `ApiIdentifier` as parent information and `Shortname as child information and if both are the same we know that is an Endpoint.

**Comment by gappc** [comment]
@sseppi yes we can do that.

Please note that we have discussion #307, issue #278 and probably many more places that discuss MetaData API, menu entries, deprecation and so on, which are all related. It is hard to follow the requirements and decisions. Could you please try to identify and bring together duplicate issues and discussions to simplify the overview?

**Comment by sseppi** [comment]
> @sseppi yes we can do that.
> 
> Please note that we have discussion #307, issue #183, #184, #278 and probably many more places that discuss MetaData API, menu entries, deprecation and so on, which are all related. It is hard to follow the requirements and decisions. Could you please try to identify and bring together duplicate issues and discussions to simplify the overview?

@gappc The discussion #307 (How to show deprecated dataset) and the issue #183 (Dataset-Overview) and #184 (Modify Metadata) concern to different features that involve the Metadata API but that I see all as different issues since every issue concerns to different functionalities that has to be implemented also in different iteration.

The only duplicated issue is this one with the #278. I will copy in this issue the contents about the navigation we discussed in the issue #278 and close the other issue.

**Comment by sseppi** [comment]
One first feedback I got from the first beta tester is that there are too many views for each dataset and this is confusing the users.

In my point of view, while we plan the optimization of the users we need to consider the following questions:
- can we fuse the Edit and Detal view and activate the edit through a toggle?
- how do we deal with the Quick View?

Another aspect that we have to consider is how do we connect the Data Browser with all other Open Data Hub tools as for example:
- Web Component Store: https://webcomponents.opendatahub.com/
- Documentation: https://docs.opendatahub.com
- Analytics: https://analytics.opendatahub.com/
- etc.

Probably can be integrated by using the Dataset Overview for example by adding there 
- the link to the Web Components that integrates the data of the dataset;
- the link to the dataset specific documentation, tutorials, etc.
- the link to the analytics dashboard that shows the data.

For Analytics we have to consider also how we want integrate the tool once we will migrate the mobility domain.

**Comment by sseppi** [comment]
@pkritzinger @Mazzolintis 

I share with you the link to the platform we saw today during the meeting:
https://www.tourismdata.ch/

**Comment by gappc** [comment]
@sseppi I overlooked your question regarding the label

I suggest to take the label value from an environment variable during compile time (like the URL for KeyCloak). That way we are free to put there whatever we want for each environment.

**Comment by sseppi** [comment]
Hi @gappc ,

> One question: could we put the "Dataset filter and search option" part into it's own issue? This issue here get's a bit crowded by the different tasks and I'm starting to lose the oversight (don't know how it's for you).

Yes, at the next Refinement Meeting, with a more clear overview of the task that will be included in this issue, we can decide all together how to split this issue in smaller task

> I suggest to take the label value from an environment variable during compile time (like the URL for KeyCloak). That way we are free to put there whatever we want for each environment.

Super, fine for me. Now we need only the labels that @pkritzinger and @Mazzolintis are preparing.

**Comment by pkritzinger** [comment]
hi @sseppi @gappc 
Please find the updated wireframes here:
https://www.figma.com/file/ue1rCbsC5gbrXhzqJKICW5/Concept-Filter-%26-Sort?type=whiteboard&node-id=179%3A814&t=X2SID699oxNIkk45-1

- Blue remarks are of general character
- Grey remarks are topics that emerged during the last meeting
- Green remarks are the answer to grey remarks

One comment - @sseppi wrote:
>allow to combine more than one filter option applying the AND operator between the filter options of the same filters group and the OR operator between the filter options of different filters groups

I understood it the other way round, i.e. Within Categories (e.g. Domain) Or-Connection of Filters and Between Categories (e.g. Tags, Domain) And-Connection of Filters. Let's discuss :)

Further I would like to add one thing: we discussed about the possibility to hide/ deactivate filter options that if applied lead to 0 results.

**Comment by pkritzinger** [comment]
>The filter in the screenshot below will include list the datasets that that are satisfying the following requirement
[(Tourism OR Mobility) AND (Weather OR Culture)]

Nice @sseppi we're on the same track 😄

**Comment by sseppi** [comment]
**Homepage**:
- the information of the number of dataset related to each domain will be taken only from the Metadata API and the loading bar will be put only instead of the dataset number;
- the video won't be played in autoplay.
- for the contributor the footer will link to the community section of the Open Data Hub website.
- the correct naming of the tool we choose is: Data Browser;
- all the links has been defined directly in the Figma file.

**Dataset**:
- the ``Apply button`` will be removed since the filters will be applied as soon as selected by the user;
- the ``Reset all filter`` will be put on top and at the bottom of the filter list;
- the ``Show newest`` will be defined in the next iteration and in the meantime the implementation will be done without;
- the ``Show deprecated`` per default is turned off and when it is turned on it adds the deprecated datasets too;
- the search button will be kept and the user must click on it to start the search;
- the mobile version of the filters will be proposed by @pkritzinger and it will contain the notification of the active filter;
- search and filters will work combined.

**Comment by gappc** [comment]
@sseppi do you have a link to the video?

**Comment by sseppi** [comment]
> @sseppi do you have a link to the video?

We need to create the video, we need to identify one or produce one.

@pkritzinger what about putting a static image fro the moment and then change it to a video one we have the video ready?

**Comment by pkritzinger** [comment]
@sseppi perfectly fine. We will make a proposal for an image.

**Comment by sseppi** [comment]
@pkritzinger this morning, in an internal meeting with @henri-egger and @emilylouisesimonis , we noticed a few small things we would change in the footer design you proposed.
I put the list here below:

- For the three link in the footer we would like to change the ordering as follows (from left to right):
  1.  Data Browser (left)
  2. opendatahub.com (middle)
  3. NOI website (right)
- We would like to change the "Other ODH Topics" into "Quickstart"

Then we tried to decline the design also for the Open Data Hub website, and there we need other small changes to be applied only on the website:
- the link on top of the footer will be only OpenDataHub.com and NOI.bz.it;

Finally we noticed that in the mobile design the it is missing the design of the menu open.

Can you please:
- update the Data Browser footer design?
- create a footer design with only two links for the Open Data Hub website
- update the design with the menu open in mobile version.

If you have questions, feel free to ask.

**Comment by pkritzinger** [comment]
@sseppi here's the updated version of the Home/ Dataset overview.
https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=608%3A6861&mode=design&t=XCXddE9TFSETLK7j-1

>create a footer design with only two links for the Open Data Hub website

In this case we would simply hide hide the respective link?

**Comment by henri-egger** [comment]
@gappc Already forgot something.

Regarding the TESTING badge, what are your thoughts about this proposal?

> @gappc I would now go about implementing the new footer and navbar design. Just a quick question regarding the new "TESTING" badge to be added. This badge will be present when viewing the databrowser in the testing environment such as testingmachine.eu. Obviously it should be decided whether or not to show the badge programmatically. When implementing the same badge for the web component store I solved this by supplying an environment variable which is set differently by the github action for test and production deploys. Are you fine with this solution?

**Comment by gappc** [comment]
@henri-egger same for me ;)

The idea was to use the `.env` file ([see example file here](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/blob/development/databrowser/.env.example)) and set a value like `VITE_APP_BADGE`, which can be overwritten during the build.

If we take that way, then please add the value also to the [env.d.ts](https://github.com/noi-techpark/it.bz.opendatahub.databrowser/blob/development/databrowser/src/env.d.ts) file to have it available with TypeScript support.

**Comment by henri-egger** [comment]
@gappc exactly, maybe it could be a comma separated value list as there will be multiple badges at once?

**Comment by pkritzinger** [comment]
@sseppi here's an update based on the discussion we had last week (25.07.23);
- Tooltip text for aggregated and single dataset
- Search without button

https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=972%3A8129&mode=design&t=RHeRAObSN41H92u5-1
